zerver/lib/actions.py

```diff
@@ -4985,23 +4986,32 @@ def get_status_dict(requesting_user_profile: UserProfile) -> Dict[str, Dict[str,
     return UserPresence.get_status_dict_by_realm(requesting_user_profile.realm_id)
 
 def get_cross_realm_dicts() -> List[Dict[str, Any]]:
+    # Using the deferred import to avoid circular import error.
+    from zerver.lib.events import format_user_row
+
     users = bulk_get_users(list(settings.CROSS_REALM_BOT_EMAILS), None,
                            base_query=UserProfile.objects.filter(
-                               realm__string_id=settings.SYSTEM_BOT_REALM)).values()
-    return [{'email': user.email,
-             'user_id': user.id,
-             'is_admin': user.is_realm_admin,
-             'is_bot': user.is_bot,
-             'avatar_url': avatar_url(user),
-             'timezone': user.timezone,
-             'date_joined': user.date_joined.isoformat(),
-             'bot_owner_id': None,
-             'full_name': user.full_name}
-            for user in users
+                           realm__string_id=settings.SYSTEM_BOT_REALM)).values()
+
+    user_dicts = {}
+    for user in users:
+        user_dicts[user.id]=vars(user)
+        user_dicts[user.id]['realm']=user.realm
+        user_dicts[user.id]['user_profile']=user
+
+    return [format_user_row(user['realm'],
+                            user_profile = user['user_profile'],
+                            row = user,
+                            client_gravatar = False,
+                            custom_profile_field_data = None,
+                            include_custom_profile_fields=False,
+                            caller= 'get_cross_realm_dicts'
+                            )
+            for user in user_dicts.values()
             # Important: We filter here, is addition to in
             # `base_query`, because of how bulk_get_users shares its
             # cache with other UserProfile caches.
-            if user.realm.string_id == settings.SYSTEM_BOT_REALM]
+            if user['realm'].string_id == settings.SYSTEM_BOT_REALM]
```

Comment:
This is likely cleaner to do using a loop appending things to a list rather than building and dict and then doing a list comprehension.